### PR TITLE
Xbox Cloud Gaming

### PR DESCRIPTION
**Pull request by marzeq**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/marzeq/xbox-cloud-gaming-electron
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build me.marzeq.xbox-cloud-gaming-electron

File: me.marzeq.xbox-cloud-gaming-electron.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Xbox Cloud Gaming
+Comment=Stream games using the Xbox Cloud Gaming service
+Type=Application
+Exec=
```
Comment (hfiguiere) [inline me.marzeq.xbox-cloud-gaming-electron.desktop:6]: exec must be set.

File: me.marzeq.xbox-cloud-gaming-electron.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+app-id: me.marzeq.xbox-cloud-gaming-electron
+branch: stable
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
Comment (hfiguiere) [inline me.marzeq.xbox-cloud-gaming-electron.yaml:4]: you should use 22.08

File: me.marzeq.xbox-cloud-gaming-electron.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    build-commands:
+      - unzip -d xbox-cloud-gaming-electron xbox-cloud-gaming-electron_*.zip
+      - rm xbox-cloud-gaming-electron_*.zip
+      - cp -r xbox-cloud-gaming-electron /app
+      - install -Dm 644 me.marzeq.xbox-cloud-gaming-electron.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm 644 me.marzeq.xbox-cloud-gaming-electron.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 644 me.marzeq.xbox-cloud-gaming-electron.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
+      - install -Dm 755 start-xbox-cloud-gaming-electron.sh /app/bin/start-xbox-cloud-gaming-electron
+    sources:
+      - type: file
```
Comment (hfiguiere) [inline me.marzeq.xbox-cloud-gaming-electron.yaml:31]: you can make this `archive` and you won't have to unzip. You can even set `dest`.

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: since you say the licence is GPL can't this be build for aarch64 too?

**Review by hfiguiere — COMMENTED** [review]
You have two icon files.

I using the XBox logo really allowed?

File: me.marzeq.xbox-cloud-gaming-electron.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+app-id: me.marzeq.xbox-cloud-gaming-electron
```
Comment (hfiguiere) [inline me.marzeq.xbox-cloud-gaming-electron.yaml:1]: There is no `marzeq.me` domain found. This mean the app-id does not meet the requirements.

File: me.marzeq.xbox-cloud-gaming-electron.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+branch: stable
```
Comment (hfiguiere) [inline me.marzeq.xbox-cloud-gaming-electron.yaml:2]:
```suggestion
```

File: flathub.json (hunk 4 above)
Comment (marzeq) [reply flathub.json:2]: I'm pretty sure electron support for ARM is a bit shaky and I don't have an ARM device to test it on

**Comment by LeandroStanger** [comment]
bot, build me.marzeq.xbox-cloud-gaming-electron

File: me.marzeq.xbox-cloud-gaming-electron.yaml (hunk 6 above)
Comment (marzeq) [reply me.marzeq.xbox-cloud-gaming-electron.yaml:2]: Why is that?

File: me.marzeq.xbox-cloud-gaming-electron.yaml (hunk 5 above)
Comment (marzeq) [reply me.marzeq.xbox-cloud-gaming-electron.yaml:1]: That's my old domain, I sometimes forget I have a new one. Will update to my current one

File: me.marzeq.xbox-cloud-gaming-electron.yaml (hunk 2 above)
Comment (marzeq) [reply me.marzeq.xbox-cloud-gaming-electron.yaml:4]: Will fix

File: me.marzeq.xbox-cloud-gaming-electron.desktop (hunk 1 above)
Comment (marzeq) [reply me.marzeq.xbox-cloud-gaming-electron.desktop:6]: Will fix

File: me.marzeq.xbox-cloud-gaming-electron.yaml (hunk 6 above)
Comment (hfiguiere) [reply me.marzeq.xbox-cloud-gaming-electron.yaml:2]: you don't set the branch. Whatever example you used as model is incorrect.

File: me.marzeq.xbox-cloud-gaming-electron.yaml (hunk 3 above)
Comment (marzeq) [reply me.marzeq.xbox-cloud-gaming-electron.yaml:31]: I'm not sure if I've done this correctly, but this should now be fixed

**Comment by marzeq** [comment]
Changed ID from me.marzeq.xbox-cloud-gaming-electron to codes.marzeq.xbox-cloud-gaming-electron (forgot I have a new domain now 😅)

**Comment by marzeq** [comment]
> You have two icon files.
> 
> I using the XBox logo really allowed?

I'm not quite sure, I'll have to check on that

File: codes.marzeq.xbox-cloud-gaming-electron.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>codes.marzeq.xbox-cloud-gaming-electron</id>
+  <launchable type="desktop-id">me.marzeq.xbox-cloud-gaming-electron.desktop</launchable>
```
Comment (hfiguiere) [inline codes.marzeq.xbox-cloud-gaming-electron.metainfo.xml:4]:
```suggestion
  <launchable type="desktop-id">codes.marzeq.xbox-cloud-gaming-electron.desktop</launchable>
```

File: codes.marzeq.xbox-cloud-gaming-electron.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
+app-id: codes.marzeq.xbox-cloud-gaming-electron
+branch: stable
```
Comment (hfiguiere) [inline codes.marzeq.xbox-cloud-gaming-electron.yaml:2]:
```suggestion
```

**Comment by marzeq** [comment]
> > You have two icon files.
> > I using the XBox logo really allowed?
> 
> I'm not quite sure, I'll have to check on that

https://www.xbox.com/en-US/legal/fansiteguidelines Nope, will change

File: codes.marzeq.xbox-cloud-gaming-electron.yaml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  - --share=network
+  - --filesystem=xdg-run/app/com.discordapp.Discord:create
+
+modules:
+  - name: xbox-cloud-gaming-electron
+    buildsystem: simple
+    build-commands:
+      - install -Dm 644 dest/codes.marzeq.xbox-cloud-gaming-electron.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm 644 dest/codes.marzeq.xbox-cloud-gaming-electron.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 644 dest/codes.marzeq.xbox-cloud-gaming-electron.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
```
Comment (hfiguiere) [inline codes.marzeq.xbox-cloud-gaming-electron.yaml:25]:
```suggestion
      - install -Dm 644 codes.marzeq.xbox-cloud-gaming-electron.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
      - install -Dm 644 codes.marzeq.xbox-cloud-gaming-electron.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
      - install -Dm 644 codes.marzeq.xbox-cloud-gaming-electron.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
```

Also you removed the actually copy of the content. That doesn't change.

File: codes.marzeq.xbox-cloud-gaming-electron.yaml [hunk 10]
```diff
@@ -0,0 +1,56 @@
+app-id: codes.marzeq.xbox-cloud-gaming-electron
+branch: stable
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
Comment (hfiguiere) [inline codes.marzeq.xbox-cloud-gaming-electron.yaml:4]:
```suggestion
runtime-version: '22.08'
```

File: codes.marzeq.xbox-cloud-gaming-electron.yaml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+          version-query: .tag_name | sub("^v"; "")
+          url-query: .assets[] | select(.name=="xbox-cloud-gaming-electron_" + $version +
+            ".zip") | .browser_download_url
+      - type: file
+        path: codes.marzeq.xbox-cloud-gaming-electron.metainfo.xml
+      - type: file
+        path: codes.marzeq.xbox-cloud-gaming-electron.png
+      - type: file
+        path: codes.marzeq.xbox-cloud-gaming-electron.desktop
+      - type: script
```
Comment (hfiguiere) [inline codes.marzeq.xbox-cloud-gaming-electron.yaml:46]: this script doesn't have a dest-filename.

Comment (hfiguiere) [reply codes.marzeq.xbox-cloud-gaming-electron.yaml:46]: maybe you meant "type: shell" to run it prior to the build.

File: codes.marzeq.xbox-cloud-gaming-electron.yaml (hunk 9 above)
Comment (marzeq) [reply codes.marzeq.xbox-cloud-gaming-electron.yaml:24]: Does the `dist` param for `type: archive` you mentioned actually change anything then?

**Comment by marzeq** [comment]
I also have a question. Does flatpak provide libva-utils and VA-API in general or do I have to add it as a dependency. My app relies on vainfo to check for the status of hardware acceleration and I'd like to know whether I have to do anything to get it working in a flatpak.

**Comment by LeandroStanger** [comment]
bot, build me.marzeq.xbox-cloud-gaming-electron

**Comment by LeandroStanger** [comment]
bot, build codes.marzeq.xbox-cloud-gaming-electron

**Comment by marzeq** [comment]
> bot, build codes.marzeq.xbox-cloud-gaming-electron

Build is broken atm, I'm waiting on a reply to https://github.com/flathub/flathub/pull/3840#discussion_r1083547252

**Comment by marzeq** [comment]
Shoudl build now

**Comment by marzeq** [comment]
```
flatpak-builder: Cleanup xbox-cloud-gaming-electronError: module xbox-cloud-gaming-electron: Error moving file /srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/build/xbox-cloud-gaming-electron-1/.uncompressSU1UY1/swiftshader/libEGL.so: File exists
```

Not sure why it fails, as the previous way of unpacking manually worked just fine

**Comment by hfiguiere** [comment]
As I said previously, you never copy the content of the archive. You removed that.

**Comment by hfiguiere** [comment]
also I'll reiterate: you have TWO icon files here. It's one too many.

Actually .desktop. icon and appstream file should be managed upstream.

**Pull request closed without merging**